### ISSUE TITLE
[codegen/nodejs] Generated provider code's pulumiType incompatible with `isInstance`

### ISSUE THREAD

**Issue by kpitzen** [pulumi/pulumi#12584]
### What happened?

This is a generalization of this reported issue on `pulumi-gcp`:
https://github.com/pulumi/pulumi-gcp/issues/1045

The tl;dr is that the provider instance's `pulumiType` is too precise to match against the module's `pulumiType` (`"pulumi:providers:gcp"` vs `gcp`). These two should be aligned.

### Expected Behavior

```typescript
gcp.Provider.isInstance(gcpProvider)
```
prints "true"

### Steps to reproduce

```typescript
import * as gcp from '@pulumi/gcp';

const gcpProvider= new gcp.Provider("default");
console.log('isInstance', gcp.Provider.isInstance(gcpProvider))
```

### Output of `pulumi about`

CLI
Version 3.55.0
Go Version go1.19.5
Go Compiler gc

Plugins
NAME VERSION
aws 4.5.1
gcp 6.50.0
kubernetes 3.21.4
nodejs unknown
postgresql 3.0.1
random 4.1.1
tls 4.10.0

Host
OS darwin
Version 12.1
Arch x86_64

This project is written in nodejs: executable='/usr/local/bin/node' version='v14.19.3'

Current Stack: ####

Found no resources associated with test

Found no pending operations associated with test

Backend
Name pulumi.com
URL https://app.pulumi.com/mleclercq
User mleclercq
Organizations mleclercq, automat-ai

Dependencies:
NAME VERSION
@types/node 12.20.46
@typescript-eslint/parser 2.6.1
eslint-plugin-prettier 3.1.1
eslint 6.6.0
prettier 1.19.1
typescript 3.9.10
@types/js-base64 2.3.2
@typescript-eslint/eslint-plugin 2.3.1
eslint-config-prettier 6.5.0
js-base64 2.6.4
nano 8.2.2

### Additional context

Linked issue:
https://github.com/pulumi/pulumi-gcp/issues/1045

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).
